From c3106657266c3df7cdc77aed87f287ec53a1befa Mon Sep 17 00:00:00 2001
From: Bruno Martins <bgcngm@gmail.com>
Date: Tue, 12 Dec 2023 16:12:32 +0000
Subject: [PATCH 87/98] Settings: Move SIM restrictions currently applied to
 per-app data usage

When Google introduced commit d3b35f2, all the custom addons for
per-app network isolation became hidden on devices that set the
overlay config_show_sim_info to "false".

Expose only the toggles that really make sense, i.e. for the mentioned
usecase, leave the main toggle (restrict all), WiFi and VPN data
toggles.

Change-Id: I4720483d98ae69fbb6eec8fa749e8d071e91c629
---
 res/xml/app_data_usage.xml                           | 8 +++++---
 src/com/android/settings/datausage/AppDataUsage.java | 5 +----
 2 files changed, 6 insertions(+), 7 deletions(-)

diff --git a/res/xml/app_data_usage.xml b/res/xml/app_data_usage.xml
index 8b2b7018d95..b7dc2b0e128 100644
--- a/res/xml/app_data_usage.xml
+++ b/res/xml/app_data_usage.xml
@@ -33,8 +33,7 @@
         settings:controller="com.android.settings.datausage.AppDataUsageSummaryController"/>
 
     <com.android.settingslib.widget.UntitledPreferenceCategory
-        android:key="app_data_usage_settings_category"
-        settings:isPreferenceVisible="@bool/config_show_sim_info">
+        android:key="app_data_usage_settings_category">
 
         <Preference
             android:key="app_settings"
@@ -54,12 +53,14 @@
         <com.android.settingslib.RestrictedSwitchPreference
             android:key="restrict_cellular"
             android:title="@string/data_usage_app_restrict_mobile"
-            android:summary="@string/data_usage_app_restrict_mobile_summary" />
+            android:summary="@string/data_usage_app_restrict_mobile_summary"
+            settings:isPreferenceVisible="@bool/config_show_sim_info" />
 
         <com.android.settingslib.RestrictedSwitchPreference
             android:key="restrict_background"
             android:title="@string/data_usage_app_restrict_background"
             android:summary="@string/data_usage_app_restrict_background_summary"
+            settings:isPreferenceVisible="@bool/config_show_sim_info"
             settings:useAdditionalSummary="true"
             settings:restrictedSwitchSummary="@string/disabled_by_admin" />
 
@@ -72,6 +73,7 @@
             android:key="unrestricted_data_saver"
             android:title="@string/unrestricted_app_title"
             android:summary="@string/unrestricted_app_summary"
+            settings:isPreferenceVisible="@bool/config_show_sim_info"
             settings:useAdditionalSummary="true"
             settings:restrictedSwitchSummary="@string/disabled_by_admin" />
 
diff --git a/src/com/android/settings/datausage/AppDataUsage.java b/src/com/android/settings/datausage/AppDataUsage.java
index ab689f8e921..88db732a1f5 100644
--- a/src/com/android/settings/datausage/AppDataUsage.java
+++ b/src/com/android/settings/datausage/AppDataUsage.java
@@ -153,7 +153,7 @@ public class AppDataUsage extends DataUsageBaseFragment implements OnPreferenceC
         final UidDetailProvider uidDetailProvider = getUidDetailProvider();
 
         if (mAppItem.key > 0) {
-            if ((!Utils.isMobileDataCapable(mContext)) || !UserHandle.isApp(mAppItem.key)) {
+            if (!UserHandle.isApp(mAppItem.key)) {
                 final UidDetail uidDetail = uidDetailProvider.getUidDetail(mAppItem.key, true);
                 mIcon = uidDetail.icon;
                 mLabel = uidDetail.label;
@@ -327,9 +327,6 @@ public class AppDataUsage extends DataUsageBaseFragment implements OnPreferenceC
     private void updatePrefs(boolean restrictBackground, boolean unrestrictData,
             boolean restrictAll, boolean restrictCellular, boolean restrictVpn,
             boolean restrictWifi, boolean hasInternetPermission) {
-        if (!Utils.isMobileDataCapable(mContext)) {
-            return;
-        }
         setBackPreferenceListAnimatorIfLoaded();
         final EnforcedAdmin admin = RestrictedLockUtilsInternal
                 .checkIfMeteredDataUsageUserControlDisabled(mContext, mPackageName,
